python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1155,6 +1209,19 @@ def submit_run_request(
     instance.submit_run(run.run_id, workspace)
 
 
+def _execution_plan_targets_asset_selection(
+    execution_plan_snapshot: ExecutionPlanSnapshot, asset_selection: Sequence[AssetKey]
+) -> bool:
+    output_asset_keys = set()
+    for step in execution_plan_snapshot.steps:
+        if step.key in execution_plan_snapshot.step_keys_to_execute:
+            for output in step.outputs:
+                asset_key = check.not_none(output.properties).asset_key
+                if asset_key:
+                    output_asset_keys.add(asset_key)
+    return all(key in output_asset_keys for key in asset_selection)
```

Comment:
this is the main part I wouldn't mind @OwenKephart or @sryza to sanity check - I believe that this always catches what we want with no false positives, but I can't conclusively say there's no weird edge case where this results in a false positive of some sort

Reply:
I feel comfortable with this bit, as it's essentially a dupe of the fairly battle-tested asset_materialization_planned code